### ISSUE TITLE
Crash: +[QNMapper fillProduct:]

### ISSUE THREAD

**Issue by HassanTaleb90** [qonversion/qonversion-ios-sdk#507]
<img width="804" alt="Screenshot 2024-05-04 at 4 59 51 AM" src="https://github.com/qonversion/qonversion-ios-sdk/assets/49780321/2a968374-f8d3-43a9-84cb-c690c88acfd8">

<img width="1349" alt="Screenshot 2024-05-04 at 5 00 35 AM" src="https://github.com/qonversion/qonversion-ios-sdk/assets/49780321/1e1d8175-bfc8-4654-aaaa-38375a00775e">

Crash log:
```
Incident Identifier: D781E365-06E4-4DF2-95BE-C7E8833C0AD8
Hardware Model:      iPhone14,2
Process:             mockup [1594]
Path:                /private/var/containers/Bundle/Application/1995F8FF-A5D1-4EE7-9497-9A497A5E2C13/mockup.app/mockup
Identifier:          ...
Version:             2.4 (2)
AppStoreTools:       15E204
AppVariant:          1:iPhone14,2:16
Code Type:           ARM-64 (Native)
Role:                Foreground
Parent Process:      launchd [1]
Coalition:           ...

Date/Time:           2024-04-26 12:09:40.6298 +0300
Launch Time:         2024-04-26 12:09:40.2154 +0300
OS Version:          iPhone OS 16.6.1 (20G81)
Release Type:        User
Baseband Version:    2.80.01
Report Version:      104

Exception Type:  EXC_CRASH (SIGABRT)
Exception Codes: 0x0000000000000000, 0x0000000000000000
Termination Reason: SIGNAL 6 Abort trap: 6
Terminating Process: mockup [1594]

Triggered by Thread:  5

Last Exception Backtrace:
0   CoreFoundation                	0x1ab8a8cb4 __exceptionPreprocess + 164 (NSException.m:202)
1   libobjc.A.dylib               	0x1a493c3d0 objc_exception_throw + 60 (objc-exception.mm:356)
2   CoreFoundation                	0x1aba1dab8 -[NSObject(NSObject) doesNotRecognizeSelector:] + 136 (NSObject.m:140)
3   CoreFoundation                	0x1ab8bf0e8 ___forwarding___ + 1592 (NSForwarding.m:3578)
4   CoreFoundation                	0x1ab925900 _CF_forwarding_prep_0 + 96 (:-1)
5   mockup                        	0x104f52ecc +[QNMapper fillProduct:] + 72 (QNMapper.m:269)
6   mockup                        	0x104f51be8 +[QNMapper fillProductsToArray:offeringID:] + 196 (QNMapper.m:106)
7   mockup                        	0x104f519d4 +[QNMapper fillProducts:] + 120 (QNMapper.m:87)
8   mockup                        	0x104f51588 +[QNMapper fillLaunchResult:] + 528 (QNMapper.m:37)
9   mockup                        	0x104f4aac0 __33-[QNProductCenterManager launch:]_block_invoke + 760 (QNProductCenterManager.m:775)
10  mockup                        	0x104f549ac __45-[QNAPIClient processDictRequest:completion:]_block_invoke + 100 (QNAPIClient.m:90)
11  mockup                        	0x104f54ccc __55-[QNAPIClient processRequest:parseResponse:completion:]_block_invoke + 216 (QNAPIClient.m:114)
12  mockup                        	0x104f57e8c __69-[QNAPIClient dataTaskWithRequest:tryCount:parseResponse:completion:]_block_invoke + 836 (QNAPIClient.m:0)
13  CFNetwork                     	0x1ac8d109c __40-[__NSURLSessionLocal taskForClassInfo:]_block_invoke + 484 (LocalSession.mm:718)
14  CFNetwork                     	0x1ac8e0bd8 __49-[__NSCFLocalSessionTask _task_onqueue_didFinish]_block_invoke_2 + 164 (LocalSessionTask.mm:544)
15  libdispatch.dylib             	0x1b2d6e320 _dispatch_call_block_and_release + 32 (init.c:1518)
16  libdispatch.dylib             	0x1b2d6feac _dispatch_client_callout + 20 (object.m:560)
17  libdispatch.dylib             	0x1b2d77534 _dispatch_lane_serial_drain + 668 (queue.c:3885)
18  libdispatch.dylib             	0x1b2d780d8 _dispatch_lane_invoke + 436 (queue.c:3976)
19  libdispatch.dylib             	0x1b2d82cdc _dispatch_workloop_worker_thread + 648 (queue.c:6507)
20  libsystem_pthread.dylib       	0x20b6d9ddc _pthread_wqthread + 288 (pthread.c:2618)
21  libsystem_pthread.dylib       	0x20b6d9b7c start_wqthread + 8 (:-1)

Thread 0 name:
Thread 0:
0   libz.1.dylib                  	0x0000000232550f38 inflate_fast + 420 (inffast.c:163)
1   ImageIO                       	0x00000001b0ae19d4 png_read_IDAT_data + 188 (pngrutil.c:4496)
2   ImageIO                       	0x00000001b0cc2440 png_read_row_indexed + 200 (pngread.c:635)
3   ImageIO                       	0x00000001b0b07bf8 PNGReadPlugin::DecodeFrameStandard(IIOImageReadSession*, ReadPluginData const&, PNGPluginData const&, IIODecodeFrameParams&) + 1504 (PNG_readPlugin.cpp:4390)
4   ImageIO                       	0x00000001b0acb4e8 PNGReadPlugin::DecodeUncomposedFrames(IIOImageRead*, GlobalPNGInfo*, ReadPluginData const&, PNGPluginData const&, std::__1::vector<IIODecodeFrameParams, std::__1::allocator<IIODecodeFrameParams> >&) + 112 (PNG_readPlugin.cpp:4684)
5   ImageIO                       	0x00000001b0acc2e8 PNGReadPlugin::copyImageBlockSetStandard(InfoRec*, CGImageProvider*, CGRect, CGSize, __CFDictionary const*) + 1360 (PNG_readPlugin.cpp:3161)
6   ImageIO                       	0x00000001b0b139c0 PNGReadPlugin::copyImageBlockSet(InfoRec*, CGImageProvider*, CGRect, CGSize, __CFDictionary const*) + 44 (PNG_readPlugin.cpp:3408)
7   ImageIO                       	0x00000001b0ac6270 IIO_Reader::CopyImageBlockSetProc(void*, CGImageProvider*, CGRect, CGSize, __CFDictionary const*) + 280 (IIOReader.cpp:1454)
8   ImageIO                       	0x00000001b0ac1d20 IIOImageProviderInfo::copyImageBlockSetWithOptions(CGImageProvider*, CGRect, CGSize, __CFDictionary const*) + 744 (CGImagePlus.cpp:2326)
9   ImageIO                       	0x00000001b0ac9df8 IIOImageProviderInfo::CopyImageBlockSetWithOptions(void*, CGImageProvider*, CGRect, CGSize, __CFDictionary const*) + 844 (CGImagePlus.cpp:2638)
10  CoreGraphics                  	0x00000001ad344988 imageProvider_retain_data + 92 (CGDataProviderImageProvider.c:91)
11  CoreGraphics                  	0x00000001ad36271c CGDataProviderRetainData + 80 (CGDataProvider.c:922)
12  CoreGraphics                  	0x00000001ad37f3a4 provider_for_destination_retain_data + 28 (CGDataProviderForDestination.c:705)
13  CoreGraphics                  	0x00000001ad36271c CGDataProviderRetainData + 80 (CGDataProvider.c:922)
14  CoreGraphics                  	0x00000001ad3416e4 CGAccessSessionCreate + 108 (CGAccessSession.c:71)
15  CoreGraphics                  	0x00000001ad30e30c img_data_lock + 2404 (CGSImage.c:4934)
16  CoreGraphics                  	0x00000001ad3536e0 CGSImageDataLock + 1328 (CGSImage.c:5847)
17  CoreGraphics                  	0x00000001ad3263f0 ripc_AcquireRIPImageData + 712 (RIPImage.c:337)
18  CoreGraphics                  	0x00000001ad343b58 ripc_DrawImage + 824 (RIPContext.c:1432)
19  CoreGraphics                  	0x00000001ad31a4c0 CGContextDrawImageWithOptions + 1120 (CGContextImage.c:303)
20  SceneKit                      	0x00000001f97885d0 C3DCreateCGBitmapContextRefFromCGImageRef + 308 (C3DImage.m:1032)
21  SceneKit                      	0x00000001f984d6fc C3DImageCacheBitmap + 224 (C3DImage.m:1470)
22  SceneKit                      	0x00000001f978ccb4 -[SCNMaterialProperty _updateC3DImageWithContents:] + 124 (SCNMaterialProperty.m:1067)
23  SceneKit                      	0x00000001f97eaa80 setupSCNMaterialProperty(SCNMaterialProperty*, MDLMaterial*, MDLMaterialSemantic, bool) + 432 (SCNModelIOPrivate.mm:3742)
24  SceneKit                      	0x00000001f97ea77c +[SCNMaterial(SCNModelIO) materialWithMDLMaterial:] + 252 (SCNModelIOPrivate.mm:3878)
25  SceneKit                      	0x00000001f97e8dcc +[SCNGeometry(SCNModelIO) geometryWithMDLMesh:] + 308 (SCNModelIOPrivate.mm:3247)
26  SceneKit                      	0x00000001f97e6cdc +[SCNNode(SCNModelIO) nodeWithMDLObject:masterObjects:sceneNodes:skinnedMeshes:skelNodesMap:asset:options:] + 736 (SCNModelIOPrivate.mm:2594)
27  SceneKit                      	0x00000001f97e6ef4 +[SCNNode(SCNModelIO) nodeWithMDLObject:masterObjects:sceneNodes:skinnedMeshes:skelNodesMap:asset:options:] + 1272 (SCNModelIOPrivate.mm:2647)
28  SceneKit                      	0x00000001f97e1aec +[SCNScene(SCNModelIO) sceneWithMDLAsset:options:] + 352 (SCNModelIOPrivate.mm:1059)
29  SceneKit                      	0x00000001f97e0d7c loadMDLAssetWithURL + 244 (SCNModelIOPrivate.mm:793)
30  SceneKit                      	0x00000001f991e1dc -[SCNSceneSource _createSceneRefWithOptions:statusHandler:] + 544 (SCNSceneSource.m:432)
31  SceneKit                      	0x00000001f978b5c0 -[SCNSceneSource _sceneWithClass:options:statusHandler:] + 1512 (SCNSceneSource.m:681)
32  SceneKit                      	0x00000001f977f660 -[SCNSceneSource sceneWithClass:options:statusHandler:] + 120 (SCNSceneSource.m:709)
33  SceneKit                      	0x00000001f991e7bc -[SCNSceneSource sceneWithClass:options:error:] + 112 (SCNSceneSource.m:787)
34  mockup                        	0x0000000104ef032c @nonobjc SCNScene.__allocating_init(named:) + 28 (<compiler-generated>:0)
35  mockup                        	0x0000000104ef032c RootVC.setupScene() + 540 (RootVC + Scene.swift:152)
36  mockup                        	0x0000000104eaf528 RootVC.viewDidLoad() + 196 (RootVC.swift:188)
37  mockup                        	0x0000000104eaf740 @objc RootVC.viewDidLoad() + 32 (<compiler-generated>:0)
38  UIKitCore                     	0x00000001adc8b8f8 -[UIViewController _sendViewDidLoadWithAppearanceProxyObjectTaggingEnabled] + 84 (UIViewController.m:2856)
39  UIKitCore                     	0x00000001ad92814c -[UIViewController loadViewIfRequired] + 712 (UIViewController.m:4277)
40  UIKitCore                     	0x00000001adbea1c8 -[UINavigationController _updateScrollViewFromViewController:toViewController:] + 124 (UINavigationController.m:7446)
41  UIKitCore                     	0x00000001adab1944 -[UINavigationController _startTransition:fromViewController:toViewController:] + 196 (UINavigationController.m:7484)
42  UIKitCore                     	0x00000001adab0df4 -[UINavigationController _startDeferredTransitionIfNeeded:] + 608 (UINavigationController.m:7708)
43  UIKitCore                     	0x00000001adab0444 -[UINavigationController __viewWillLayoutSubviews] + 96 (UINavigationController.m:8020)
44  UIKitCore                     	0x00000001adab03a8 -[UILayoutContainerView layoutSubviews] + 172 (UILayoutContainerView.m:89)
45  UIKitCore                     	0x00000001ad912420 -[UIView(CALayerDelegate) layoutSublayersOfLayer:] + 1992 (UIView.m:18682)
46  QuartzCore                    	0x00000001acdccf30 CA::Layer::layout_if_needed(CA::Transaction*) + 500 (CALayer.mm:10311)
47  QuartzCore                    	0x00000001acde04ac CA::Layer::layout_and_display_if_needed(CA::Transaction*) + 148 (CALayer.mm:2519)
48  QuartzCore                    	0x00000001acdf18d8 CA::Context::commit_transaction(CA::Transaction*, double, double*) + 444 (CAContextInternal.mm:2714)
49  QuartzCore                    	0x00000001ace20e80 CA::Transaction::commit() + 648 (CATransactionInternal.mm:432)
50  UIKitCore                     	0x00000001add9a75c __34-[UIApplication _firstCommitBlock]_block_invoke_2 + 36 (UIApplication.m:12187)
51  CoreFoundation                	0x00000001ab8e26e0 __CFRUNLOOP_IS_CALLING_OUT_TO_A_BLOCK__ + 28 (CFRunLoop.c:1805)
52  CoreFoundation                	0x00000001ab949210 __CFRunLoopDoBlocks + 364 (CFRunLoop.c:1847)
53  CoreFoundation                	0x00000001ab9190e8 __CFRunLoopRun + 848 (CFRunLoop.c:2955)
54  CoreFoundation                	0x00000001ab91e3ec CFRunLoopRunSpecific + 612 (CFRunLoop.c:3418)
55  GraphicsServices              	0x00000001e6e3435c GSEventRunModal + 164 (GSEvent.c:2196)
56  UIKitCore                     	0x00000001adcaaf58 -[UIApplication _run] + 888 (UIApplication.m:3782)
57  UIKitCore                     	0x00000001adcaabbc UIApplicationMain + 340 (UIApplication.m:5372)
58  libswiftUIKit.dylib           	0x00000001b43380b0 UIApplicationMain(_:_:_:_:) + 104 (UIKit.swift:538)
59  mockup                        	0x0000000104ed2a8c specialized static UIApplicationDelegate.main() + 28 (<compiler-generated>:14)
60  mockup                        	0x0000000104ed2a8c static AppDelegate.$main() + 28 (AppDelegate.swift:0)
61  mockup                        	0x0000000104ed2a8c main + 116
62  dyld                          	0x00000001cae50dec start + 2220 (dyldMain.cpp:1165)

Thread 1:
0   libsystem_pthread.dylib       	0x000000020b6d9b74 start_wqthread + 0 (:-1)

Thread 2:
0   libsystem_pthread.dylib       	0x000000020b6d9b74 start_wqthread + 0 (:-1)

Thread 3:
0   libsystem_pthread.dylib       	0x000000020b6d9b74 start_wqthread + 0 (:-1)

Thread 4 name:
Thread 4:
0   libsystem_kernel.dylib        	0x00000001ea85eca4 mach_msg2_trap + 8 (:-1)
1   libsystem_kernel.dylib        	0x00000001ea871b74 mach_msg2_internal + 80 (mach_msg.c:201)
2   libsystem_kernel.dylib        	0x00000001ea871e4c mach_msg_overwrite + 540 (mach_msg.c:0)
3   libsystem_kernel.dylib        	0x00000001ea85f1e8 mach_msg + 24 (mach_msg.c:323)
4   CoreFoundation                	0x00000001ab918024 __CFRunLoopServiceMachPort + 160 (CFRunLoop.c:2622)
5   CoreFoundation                	0x00000001ab919250 __CFRunLoopRun + 1208 (CFRunLoop.c:3005)
6   CoreFoundation                	0x00000001ab91e3ec CFRunLoopRunSpecific + 612 (CFRunLoop.c:3418)
7   Foundation                    	0x00000001a5b92fb4 -[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 212 (NSRunLoop.m:373)
8   Foundation                    	0x00000001a5b92e9c -[NSRunLoop(NSRunLoop) runUntilDate:] + 64 (NSRunLoop.m:420)
9   UIKitCore                     	0x00000001addddcc8 -[UIEventFetcher threadMain] + 416 (UIEventFetcher.m:1376)
10  Foundation                    	0x00000001a5bac524 __NSThread__start__ + 716 (NSThread.m:963)
11  libsystem_pthread.dylib       	0x000000020b6da6b8 _pthread_start + 148 (pthread.c:893)
12  libsystem_pthread.dylib       	0x000000020b6d9b88 thread_start + 8 (:-1)

Thread 5 name:
Thread 5 Crashed:
0   libsystem_kernel.dylib        	0x00000001ea865578 __pthread_kill + 8 (:-1)
1   libsystem_pthread.dylib       	0x000000020b6e0118 pthread_kill + 268 (pthread.c:1670)
2   libsystem_c.dylib             	0x00000001b2dd0178 abort + 180 (abort.c:118)
3   libc++abi.dylib               	0x000000020b61ebf8 abort_message + 132 (:-1)
4   libc++abi.dylib               	0x000000020b60e444 demangling_terminate_handler() + 348 (:-1)
5   libobjc.A.dylib               	0x00000001a4941ea4 _objc_terminate() + 144 (objc-exception.mm:498)
6   libc++abi.dylib               	0x000000020b61dfbc std::__terminate(void (*)()) + 16 (:-1)
7   libc++abi.dylib               	0x000000020b61df60 std::terminate() + 56 (:-1)
8   libdispatch.dylib             	0x00000001b2d6fec0 _dispatch_client_callout + 40 (object.m:563)
9   libdispatch.dylib             	0x00000001b2d77534 _dispatch_lane_serial_drain + 668 (queue.c:3885)
10  libdispatch.dylib             	0x00000001b2d780d8 _dispatch_lane_invoke + 436 (queue.c:3976)
11  libdispatch.dylib             	0x00000001b2d82cdc _dispatch_workloop_worker_thread + 648 (queue.c:6507)
12  libsystem_pthread.dylib       	0x000000020b6d9ddc _pthread_wqthread + 288 (pthread.c:2618)
13  libsystem_pthread.dylib       	0x000000020b6d9b7c start_wqthread + 8 (:-1)

Thread 6:
0   libsystem_pthread.dylib       	0x000000020b6d9b74 start_wqthread + 0 (:-1)

Thread 7:
0   libsystem_pthread.dylib       	0x000000020b6d9b74 start_wqthread + 0 (:-1)

Thread 8:
0   libsystem_kernel.dylib        	0x00000001ea85ec20 semaphore_wait_trap + 8 (:-1)
1   libtbb.dylib                  	0x0000000212eebd1c tbb::internal::rml::private_worker::run() + 852 (private_server.cpp:273)
2   libtbb.dylib                  	0x0000000212eeb9bc tbb::internal::rml::private_worker::thread_routine(void*) + 16 (private_server.cpp:219)
3   libsystem_pthread.dylib       	0x000000020b6da6b8 _pthread_start + 148 (pthread.c:893)
4   libsystem_pthread.dylib       	0x000000020b6d9b88 thread_start + 8 (:-1)

Thread 9:
0   libsystem_kernel.dylib        	0x00000001ea85ec20 semaphore_wait_trap + 8 (:-1)
1   libtbb.dylib                  	0x0000000212eebd1c tbb::internal::rml::private_worker::run() + 852 (private_server.cpp:273)
2   libtbb.dylib                  	0x0000000212eeb9bc tbb::internal::rml::private_worker::thread_routine(void*) + 16 (private_server.cpp:219)
3   libsystem_pthread.dylib       	0x000000020b6da6b8 _pthread_start + 148 (pthread.c:893)
4   libsystem_pthread.dylib       	0x000000020b6d9b88 thread_start + 8 (:-1)

Thread 10:
0   libsystem_kernel.dylib        	0x00000001ea85ec20 semaphore_wait_trap + 8 (:-1)
1   libtbb.dylib                  	0x0000000212eebd1c tbb::internal::rml::private_worker::run() + 852 (private_server.cpp:273)
2   libtbb.dylib                  	0x0000000212eeb9bc tbb::internal::rml::private_worker::thread_routine(void*) + 16 (private_server.cpp:219)
3   libsystem_pthread.dylib       	0x000000020b6da6b8 _pthread_start + 148 (pthread.c:893)
4   libsystem_pthread.dylib       	0x000000020b6d9b88 thread_start + 8 (:-1)

Thread 11:
0   libsystem_kernel.dylib        	0x00000001ea85ec20 semaphore_wait_trap + 8 (:-1)
1   libtbb.dylib                  	0x0000000212eebd1c tbb::internal::rml::private_worker::run() + 852 (private_server.cpp:273)
2   libtbb.dylib                  	0x0000000212eeb9bc tbb::internal::rml::private_worker::thread_routine(void*) + 16 (private_server.cpp:219)
3   libsystem_pthread.dylib       	0x000000020b6da6b8 _pthread_start + 148 (pthread.c:893)
4   libsystem_pthread.dylib       	0x000000020b6d9b88 thread_start + 8 (:-1)

Thread 12:
0   libsystem_kernel.dylib        	0x00000001ea85ec20 semaphore_wait_trap + 8 (:-1)
1   libtbb.dylib                  	0x0000000212eebd1c tbb::internal::rml::private_worker::run() + 852 (private_server.cpp:273)
2   libtbb.dylib                  	0x0000000212eeb9bc tbb::internal::rml::private_worker::thread_routine(void*) + 16 (private_server.cpp:219)
3   libsystem_pthread.dylib       	0x000000020b6da6b8 _pthread_start + 148 (pthread.c:893)
4   libsystem_pthread.dylib       	0x000000020b6d9b88 thread_start + 8 (:-1)

Thread 13:
0   libsystem_kernel.dylib        	0x00000001ea85f164 __semwait_signal + 8 (:-1)
1   libsystem_c.dylib             	0x00000001b2db8888 nanosleep + 220 (nanosleep.c:104)
2   libc++.1.dylib                	0x00000001b9f06a20 std::__1::this_thread::sleep_for(std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000000000l> > const&) + 84 (thread.cpp:109)
3   libusd_ms.dylib               	0x00000002124a2fb4 void* std::__1::__thread_proxy[abi:v15006]<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, pxrInternal_v0_21__aapl__pxrReserve... + 76
4   libsystem_pthread.dylib       	0x000000020b6da6b8 _pthread_start + 148 (pthread.c:893)
5   libsystem_pthread.dylib       	0x000000020b6d9b88 thread_start + 8 (:-1)

Thread 14 name:
Thread 14:
0   libsystem_kernel.dylib        	0x00000001ea85eca4 mach_msg2_trap + 8 (:-1)
1   libsystem_kernel.dylib        	0x00000001ea871b74 mach_msg2_internal + 80 (mach_msg.c:201)
2   libsystem_kernel.dylib        	0x00000001ea871e4c mach_msg_overwrite + 540 (mach_msg.c:0)
3   libsystem_kernel.dylib        	0x00000001ea85f1e8 mach_msg + 24 (mach_msg.c:323)
4   CoreFoundation                	0x00000001ab918024 __CFRunLoopServiceMachPort + 160 (CFRunLoop.c:2622)
5   CoreFoundation                	0x00000001ab919250 __CFRunLoopRun + 1208 (CFRunLoop.c:3005)
6   CoreFoundation                	0x00000001ab91e3ec CFRunLoopRunSpecific + 612 (CFRunLoop.c:3418)
7   CFNetwork                     	0x00000001acb096f0 +[__CFN_CoreSchedulingSetRunnable _run:] + 392 (CoreSchedulingSet.mm:1562)
8   Foundation                    	0x00000001a5bac524 __NSThread__start__ + 716 (NSThread.m:963)
9   libsystem_pthread.dylib       	0x000000020b6da6b8 _pthread_start + 148 (pthread.c:893)
10  libsystem_pthread.dylib       	0x000000020b6d9b88 thread_start + 8 (:-1)


Thread 5 crashed with ARM Thread State (64-bit):
    x0: 0x0000000000000000   x1: 0x0000000000000000   x2: 0x0000000000000000   x3: 0x0000000000000000
    x4: 0x000000020b622647   x5: 0x000000016b216850   x6: 0x000000000000006e   x7: 0x0000000000000000
    x8: 0x259a1f4e90ed9941   x9: 0x259a1f4ffbcce941  x10: 0x0000000000000200  x11: 0x000000000000000b
   x12: 0x000000000000000b  x13: 0x00000000001ff800  x14: 0x00000000000007fb  x15: 0x000000009600087f
   x16: 0x0000000000000148  x17: 0x000000016b217000  x18: 0x0000000000000000  x19: 0x0000000000000006
   x20: 0x0000000000003c03  x21: 0x000000016b2170e0  x22: 0x0000000000000000  x23: 0x000000016b2170e0
   x24: 0x0000000283fac4e8  x25: 0x0000000000000000  x26: 0x0000000000000114  x27: 0x0000000000000000
   x28: 0x0000000280bbd980   fp: 0x000000016b2167c0   lr: 0x000000020b6e0118
    sp: 0x000000016b2167a0   pc: 0x00000001ea865578 cpsr: 0x40001000
   esr: 0x56000080  Address size fault


Binary Images:
        0x104ea0000 -         0x104fd7fff mockup arm64  <b2f8f8132fa63e7c8927235cec3b6cbc> /private/var/containers/Bundle/Application/1995F8FF-A5D1-4EE7-9497-9A497A5E2C13/mockup.app/mockup
        0x106f24000 -         0x106f2ffff libobjc-trampolines.dylib arm64e  <7e77fc541c3331c086e81fb7bf12836c> /private/preboot/Cryptexes/OS/usr/lib/libobjc-trampolines.dylib
        0x1a4924000 -         0x1a4967f9f libobjc.A.dylib arm64e  <eb7faf215c9f37848907affa6d92bc3b> /usr/lib/libobjc.A.dylib
        0x1a5b51000 -         0x1a642efff Foundation arm64e  <1715de5d08933af0b0c0550bb14f91ec> /System/Library/Frameworks/Foundation.framework/Foundation
        0x1ab89f000 -         0x1abc86fff CoreFoundation arm64e  <a900b4590127379e9cba0eab9c5d559f> /System/Library/Frameworks/CoreFoundation.framework/CoreFoundation
        0x1ac8b1000 -         0x1acc7bfff CFNetwork arm64e  <045d4271e64e3229a49fcaff707aa77f> /System/Library/Frameworks/CFNetwork.framework/CFNetwork
        0x1acdc3000 -         0x1ad13cfff QuartzCore arm64e  <8682cc56125a35658075978cc718b93f> /System/Library/Frameworks/QuartzCore.framework/QuartzCore
        0x1ad302000 -         0x1ad90dfff CoreGraphics arm64e  <d7f3902639023b29b5cf07c76d5e5047> /System/Library/Frameworks/CoreGraphics.framework/CoreGraphics
        0x1ad90e000 -         0x1af132fff UIKitCore arm64e  <7d57a1d1856f338d97db880c4ec8b02e> /System/Library/PrivateFrameworks/UIKitCore.framework/UIKitCore
        0x1b0aac000 -         0x1b0f0cfff ImageIO arm64e  <7ff3e4fedb953a36b968359693fdbfce> /System/Library/Frameworks/ImageIO.framework/ImageIO
        0x1b2d6c000 -         0x1b2db2fff libdispatch.dylib arm64e  <5d16936b4e4c3276ba7a69c9bc760aba> /usr/lib/system/libdispatch.dylib
        0x1b2db3000 -         0x1b2e30ff3 libsystem_c.dylib arm64e  <073251d1d4c937748cee75210751a5d8> /usr/lib/system/libsystem_c.dylib
        0x1b4303000 -         0x1b437afff libswiftUIKit.dylib arm64e  <d24b70db91ea38e9b1e0db9d15f6b17e> /usr/lib/swift/libswiftUIKit.dylib
        0x1b9ee6000 -         0x1b9f72ff7 libc++.1.dylib arm64e  <73d4f57ea1293e3794408eeb0d608273> /usr/lib/libc++.1.dylib
        0x1cae3b000 -         0x1caec0a53 dyld arm64e  <c3fc2ee4367f3086beb8420ad442bf88> /usr/lib/dyld
        0x1e6e33000 -         0x1e6e3bfff GraphicsServices arm64e  <cd8d719514e13c7ca8c950ccea9f2a17> /System/Library/PrivateFrameworks/GraphicsServices.framework/GraphicsServices
        0x1ea85e000 -         0x1ea895fe7 libsystem_kernel.dylib arm64e  <681b06a07f6b3fa3a2ce063dc1da7b1b> /usr/lib/system/libsystem_kernel.dylib
        0x1f9758000 -         0x1f9c3dfff SceneKit arm64e  <5e073ae5ed85383b9b63d9371d1dde64> /System/Library/Frameworks/SceneKit.framework/SceneKit
        0x20b60d000 -         0x20b624fff libc++abi.dylib arm64e  <1e834588aca6327a9a965aa8b13a2ae2> /usr/lib/libc++abi.dylib
        0x20b6d9000 -         0x20b6e4ff3 libsystem_pthread.dylib arm64e  <aa302f5affe5327aa0033b288b935de1> /usr/lib/system/libsystem_pthread.dylib
        0x2123eb000 -         0x212ee0fff libusd_ms.dylib arm64e  <7e3fac2a703236d7877ca52e502133b9> /usr/lib/usd/libusd_ms.dylib
        0x212ee1000 -         0x212f00fff libtbb.dylib arm64e  <6e22a886905632629ef2af4176357a0e> /usr/lib/usd/libtbb.dylib
        0x23254c000 -         0x23255dff7 libz.1.dylib arm64e  <5c731638521a30029d5515ee35048600> /usr/lib/libz.1.dylib

EOF
```

**Comment by SpertsyanKM** [comment]
Hello, @HassanTaleb90! Thank you for the report! Could you please clarify which SDK version do you use and do you know the steps to reproduce the crash? Or maybe some information about when does it happen?

**Comment by HassanTaleb90** [comment]
SDK version 5.8.4
I don't have any other information.

**Comment by SpertsyanKM** [comment]
Okay, thank you! We will try to investigate this issue

**Comment by suriksarkisyan** [comment]
Hi @HassanTaleb90 .
We updated our SDK and [released 5.12.2](https://github.com/qonversion/qonversion-ios-sdk/releases/tag/5.12.2) a fix for this issue. I hope this fix will help. Let's keep in touch, and let me know if you face it again!
